### DISCUSSION TITLE
Need help w/ parsing datetime effectively with mapstructures in go.

### DISCUSSION THREAD

**Discussion by pvbhanuteja** [pocketbase/pocketbase#4115] [Q&A]
Current setup 
```
type Usage struct {
 // usage struct
}

type UserProfile struct {
	APIKey         string   `json:"apiKey"`
	BillingAddress string   `json:"billing_address"`
	CollectionID   string   `json:"collectionId"`
	CollectionName string   `json:"collectionName"`
	Created        DateTime `json:"created"`
	ID             string   `json:"id"`
	NextPeriod     DateTime `json:"next_period"`
	PaymentMethod  string   `json:"payment_method"`
	Plan           string   `json:"plan"`
	SubID          string   `json:"sub_id"`
	Updated        DateTime `json:"string"`
	Usage          Usage    `json:"usage"`
	User           string   `json:"user"`
}

type DateTime struct {
	t time.Time
}
```

I'm trying to parse the pb api response with mapstructures

```
	response, err := collection.List(pocketbase.ParamsList{
		Size:    1,
		Page:    1,
		Filters: "apiKey~'" + apikey + "'",
	})
	fmt.Println(response)
	if err != nil {
		log.Fatal(err)
		return nil, err
	}

	if len(response.Items) == 0 {
		return nil, errors.New("No user found")
	}

	merr := mapstructure.Decode(response.Items[0], u)

	if merr != nil {
		fmt.Println(merr)
		log.Fatal(err)
		return nil, err
	}
```

And I keep on getting any effective way to parse these date strings ?
2 error(s) decoding:

* 'Created' expected a map, got 'string'
* 'Updated' expected a map, got 'string'

**Comment by ganigeorgiev** [comment]
I don't know what `collection.List` or `pocketbase.ParamsList` is.

Could you elaborate a little more on what you are trying to do and/or provide a more detailed code sample?

**Reply by ganigeorgiev** [reply]
Also note that the `mapstructure` error should be expected and it is not PocketBase specific if you are trying to decode a plain string value in your custom `DateTime` struct (unless you implement your own [UnmarshalJSON](https://pkg.go.dev/encoding/json#Unmarshaler) method).

**Reply by pvbhanuteja** [reply]
> I don't know what `collection.List` or `pocketbase.ParamsList` is.
> 
> Could you elaborate a little more on what you are trying to do and/or provide a more detailed code sample?

It's from here https://github.com/pluja/pocketbase

Basically the API response collection list as mapped in go

**Reply by ganigeorgiev** [reply]
I'm not familiar with the above library, but as mentioned earlier, you can't unmarshal a plain string value into a custom struct without specifying your own unmarshaler that will map it to the specific fields (in your case `DateTime.t`).

Alternatively, if you want to use the PocketBase `DateTime` type you can try to import [`types.DateTime`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.20.5/tools/types#DateTime).

If you have further questions related to the above library, please open them in the related repo.

**Reply by pvbhanuteja** [reply]
Did this 
```
package dbhelpers

import (
	pbtypes "github.com/pocketbase/pocketbase/tools/types"
)

type Req struct {

}

type Usage struct {
// Usage heer
}

type UserProfile struct {
	APIKey         string           `json:"apiKey"`
	BillingAddress string           `json:"billing_address"`
	CollectionID   string           `json:"collectionId"`
	CollectionName string           `json:"collectionName"`
	Created        pbtypes.DateTime `json:"created"`
	ID             string           `json:"id"`
	NextPeriod     pbtypes.DateTime `json:"next_period"`
	PaymentMethod  string           `json:"payment_method"`
	Plan           string           `json:"plan"`
	SubID          string           `json:"sub_id"`
	Updated        pbtypes.DateTime `json:"string"`
	Usage          Usage            `json:"usage"`
	User           string           `json:"user"`
}
```

But I still get same issue. 

```
{1 1 1 1 [map[apiKey:q17boyriyq05bsnactgkj billing_address: collectionId:kb3cnjx5ycujn8k collectionName:user_profile created:2024-01-01 06:30:49.308Z id:4fdwftlj54i3gyn next_period:2024-02-03 06:16:52.000Z payment_method: plan:paid sub_id:y6zx8dczq38gmc9 updated:2024-01-08 16:33:44.329Z usage:<usage> user:lw8z6vvowv2iltw]]}
2 error(s) decoding:

* 'Created' expected a map, got 'string'
* 'Updated' expected a map, got 'string'
2024/01/10 21:11:27 <nil>
```
